cpp/src/arrow/compute/exec.cc

```diff
@@ -966,12 +998,6 @@ class ScalarExecutor : public KernelExecutorImpl<ScalarKernel> {
          data_preallocated_.size() == static_cast<size_t>(output_num_buffers_ - 1) &&
          !is_nested(out_type_id) && !is_dictionary(out_type_id));
 
-    // TODO(wesm): why was this check ever here? Fixed width binary
```

Comment:
`ComputeDataPreallocate` can make sure all of the element in `data_preallocated_` could satisfy the `DCHECK`.
It's unnecessary to keep this here.

Reply:
We can keep the DCHECK you suggested before ( https://github.com/apache/arrow/pull/41975/files#r1636669846 which you suggest). In essence, they detect the same content, otherwise there will be two duplicate DCHECKs.